website/frontend/src/services/ApiService.tsx

```diff
@@ -261,9 +261,11 @@ export const subscribeToEvents = (alert: AlertContainer) => {
   const filterKey = localStorage.getItem(LOCAL_STORAGE.PUBLIC_KEY_RAW);
   nodeApi.subscribeToProgramEvents(({ method, data: { info, reason } }) => {
     // @ts-ignore
-    if (info.origin.toHex() === filterKey) {
-      const message = `${method}\n ${info.programId.toHex()}`;
+    if (data.info.origin.toHex() === filterKey) {
+      const description = method === 'InitFailure' ? reason.asDispatch.toHuman() : info.programId.toHex();
+      const message = `${method}: \n ${description}`;
```

Comment:
I think it's better to display programId in any case.
And display `asDispatch` if `reason?.isDispatched` will be true